### PR TITLE
fix(#653) Fix condition to show badge quantity

### LINKED ISSUE #653: Sharing layer availability badges don't show number or nothing at all
Currently, the bike sharing availability badge does not show the number of available bikes.

image

However, as upstream shows them:
image

For taxi, no availability badge is shown at all:

image

### PR DESCRIPTION
**Pull request by jpidelatorre**
The condition to show the amount in the badge includes a field called `state`. But that field is always null and that prevents the condition from going through.

closes #653

### CODE REVIEW

File: app/util/mapIconUtils.js [hunk 1]
```diff
@@ -1,11 +1,7 @@
 import memoize from 'lodash/memoize';
 import getSelector from './get-selector';
 import glfun from './glfun';
-import {
-  BIKESTATION_ON,
-  BIKESTATION_OFF,
-  BIKESTATION_CLOSED,
-} from './citybikes';
+import { BIKESTATION_OFF, BIKESTATION_CLOSED } from './citybikes';
```
Comment (hbruch) [inline app/util/mapIconUtils.js]: Please keep BIKESTATION_ON to stay compatible with upstream

File: app/util/mapIconUtils.js [hunk 2]
```diff
@@ -643,7 +639,7 @@ export function drawCitybikeIcon(
   }
   let color = 'green';
   if (showAvailability) {
-    if (!bikesAvailable) {
+    if (bikesAvailable < 1) {
       color = 'red';
```
Comment (hbruch) [inline app/util/mapIconUtils.js:647]: bikesAvailable might be null, so don't remove the check for that

File: app/util/mapIconUtils.js [hunk 3]
```diff
@@ -679,7 +671,7 @@ export function drawCitybikeIcon(
       tile.ctx.drawImage(image, x, y);
       x = x + width - smallCircleRadius;
       y += smallCircleRadius;
-      if (showAvailabilityBadge) {
+      if (showAvailability && bikesAvailable >= 0) {
         /* eslint-disable no-param-reassign */
```
Comment (hbruch) [inline app/util/mapIconUtils.js]: Why remove showAvailabilityBadge?

File: app/util/mapIconUtils.js [hunk 4]
```diff
@@ -667,10 +663,6 @@ export function drawCitybikeIcon(
     const smallCircleRadius = 11 * tile.scaleratio;
     x = geom.x / tile.ratio - width + smallCircleRadius * 2;
     y = geom.y / tile.ratio - height;
-    const showAvailabilityBadge =
-      showAvailability &&
-      (bikesAvailable || bikesAvailable === 0) &&
-      state === BIKESTATION_ON;
     let icon = `${iconName}_station_${color}_large`;
```
Comment (hbruch) [inline app/util/mapIconUtils.js:674]: if the reason for not showing badges is the missing state, we should include it in this check, no?

**Review by hbruch — CHANGES_REQUESTED** [review]
Please rework to stay compatible with upstream

File: app/util/mapIconUtils.js (hunk 2 above)
Comment (jpidelatorre) [reply app/util/mapIconUtils.js:647]: The value usually is `-1`. But I'm adding that condition too.

File: app/util/mapIconUtils.js (hunk 3 above)
Comment (jpidelatorre) [reply app/util/mapIconUtils.js]: It was cleaner, but now, with a more complex condition I'll put it back.

File: app/util/mapIconUtils.js (hunk 4 above)
Comment (jpidelatorre) [reply app/util/mapIconUtils.js:674]: I'm adding it back with an additional condition that allows for null states. I guess the real solution should be in the data and not in the condition.

**Comment by jpidelatorre** [comment]
@hbruch I applied all the changes you suggested.

File: app/util/mapIconUtils.js [hunk 5]
```diff
@@ -669,8 +669,8 @@ export function drawCitybikeIcon(
     y = geom.y / tile.ratio - height;
     const showAvailabilityBadge =
       showAvailability &&
-      (bikesAvailable || bikesAvailable === 0) &&
-      state === BIKESTATION_ON;
+      (Number.isSafeInteger(bikesAvailable) || bikesAvailable > -1) &&
```
Comment (hbruch) [inline app/util/mapIconUtils.js]: Why do you check for Number.isSafeInteger? With the OR criteria, negative bikesAvailable will be shown, too, no?

**Review by hbruch — CHANGES_REQUESTED** [review]
Please check again and add a screenshot of the resulting display, showing casharing, bikesharing and taxi symbol with their appearance (carsharing is expected not to have a badge)

**Comment by jpidelatorre** [comment]
As it seems, taxis don't have the badge but they have the quantity in there. This is how it looks:

image

**Comment by hbruch** [comment]
Then please investigate. They should have both, badge and number

**Comment by leonardehrenfried** [comment]
I think the reason is that we explicitly disabled this because at that time we didn't have realtime information.

**Comment by jpidelatorre** [comment]
@leonardehrenfried is right. The badge is part of the icon and there are multiple versions of the same icon in the sprite (one for each color).

image

I added the circles to all the icons.

**Review by hbruch — COMMENTED** [review]
Could you comment on L672?

File: app/util/mapIconUtils.js (hunk 5 above)
Comment (leonardehrenfried) [reply app/util/mapIconUtils.js]: Screenshot from 2021-06-18 14-46-39

It's a bit hard to see but I think -1 is rendered.

**Comment by jpidelatorre** [comment]
@hbruch Just changed the operator.

**Comment by leonardehrenfried** [comment]
Thank you!

**Pull request merged**